### ISSUE TITLE
SimplifyName: triggers on foreach where simplification is inapplicable

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#28086]
![image](https://user-images.githubusercontent.com/12466233/41805852-e489844c-7665-11e8-9ee6-c1112d96e7f8.png)

`test` is grayed out. The code action doesn't make sense (changing the variable name from `test` to `var`).

I suspect this is a regression, as I would have noticed this before otherwise.

```
Version 15.8.0 Preview 2.0
VisualStudio.15.Preview/15.8.0-pre.2.0+27729.1
```

Tagging @jinujoseph for triage.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/26952

**Comment by jinujoseph** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/26923
Fixed in 15.8.Preview3
